### PR TITLE
superset code clean up including geckodriver upgrade.

### PR DESCRIPTION
**Pull request by himanshudube97**

### CODE REVIEW

File: gensuperset/make-t4d/generate-make-t4d.sh [hunk 1]
```diff
@@ -2,15 +2,20 @@
 # Usage: ./generate-make-t4d.sh "apache/superset:3.1.arm" "tech4dev/superset:0.41" "output_folder"
 
 # Check if the correct number of arguments are provided
-if [ "$#" -ne 3 ]; then
-    echo "Usage: $0 <base_image> <output_image> <output_folder>"
+if [ "$#" -ne 4 ]; then
+    if [ "$#" -ne 4 ]; then
```
Comment (fatchat) [inline gensuperset/make-t4d/generate-make-t4d.sh:6]: this double `if` condition is a typo right

File: gensuperset/make-client/generate-make-client.sh [hunk 2]
```diff
@@ -82,22 +95,47 @@ cat <<EOF > $BUILD_SCRIPT_PATH
...
+    echo "Error: Failed to build Docker image: \$IMAGE_NAME"
+    exit 1
+fi
 EOF
 
 # Make the build script executable
 chmod +x $BUILD_SCRIPT_PATH
 
+# Generate the push script
+PUSH_SCRIPT_PATH="$OUTPUT_DIR/push.sh"
```
Comment (fatchat) [inline gensuperset/make-client/generate-make-client.sh:114]: are we pushing client's images to dockerhub? i don't think we should

Comment (himanshudube97) [reply gensuperset/make-client/generate-make-client.sh]: No, we don't push client's image to docker hub. I have removed this from the template script.

File: gensuperset/make-t4d/generate-make-t4d.sh (hunk 1 above)
Comment (himanshudube97) [reply gensuperset/make-t4d/generate-make-t4d.sh]: Yes it was a typo. Removed it .

**Review by fatchat — APPROVED** [review]

**Pull request merged**